src/main/java/core/basesyntax/service/strategy/Purchase.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.FruitInfo;
+import core.basesyntax.model.FruitType;
+import java.util.List;
+import java.util.Map;
+
+public class Purchase implements Activities {
+
+    @Override
+    public Map<FruitType, Integer> calculateBalanceAfterActivities(
+            List<FruitInfo> fruitServiceList,
+            Map<FruitType, Integer> fruitServiceMap) {
+        for (FruitInfo fruitInfo : fruitServiceList) {
+            if (fruitInfo.getActivities().equals("p")) {

```

Comment:
don't pass and iterate list
use Dao to get the fruit's quantity in DB, and save it
don't filter by activity code
only valid operations(activities) are supposed to be passed to handler. That's strategy job, to create a correspondence between handlers and operations